packages/bitcore-node/src/modules/ripple/api/csp.ts

```diff
@@ -330,6 +355,41 @@ export class RippleStateProvider extends InternalStateProvider implements IChain
       return tx;
     }
   }
+
+  async tag(
+    chain: string,
+    network: string,
+    tx: IXrpTransaction,
+    outputs: Array<ICoin> | any
+  ): Promise<{ transaction: IXrpTransaction; coins: Array<ICoin> }> {
+    const address = tx.from;
+    let involvedAddress = [address];
+    const transaction = { ...tx, wallets: new Array<ObjectId>() };
+    let coins = new Array<ICoin>();
+    if (Array.isArray(outputs)) {
+      coins = outputs.map(c => {
+        return { ...c, wallets: new Array<ObjectId>() };
+      });
+      involvedAddress.push(...coins.map(c => c.address));
+    }
+    const walletAddresses = await WalletAddressStorage.collection
+      .find({ chain, network, address: { $in: involvedAddress } })
+      .toArray();
```

Comment:
TODO: this would be better as a cursor stream

Reply:
The fetching of the walletAddresses? It's using $in, where the addresses come from one tx, so it should be a small set of addresses.